b2g/components/RemoteControlService.jsm

```diff
@@ -950,6 +990,13 @@ EventHandler.prototype = {
     });
   },
 
+  // Synthesize PIN for J-PAKE round 2 by original PIN
+  // and TLS server cert fingerprint
+  _synthesizeJPAKEPIN: function(aPIN) {
+    return SHA256(aPIN + this._server.getCertFingerprint())
+           .slice(-1 * JPAKE_PIN_LENGTH);
```

Comment:
I think the problem happens ```function toHexString(aCharCode)```. It will enlarge the size. I will pass the SHA256 value from ```finish(false)``` directly.

[0] https://dxr.mozilla.org/mozilla-central/source/netwerk/base/nsICryptoHash.idl#104
[1] https://dxr.mozilla.org/mozilla-central/source/security/manager/ssl/nsCryptoHash.cpp#420